apps/web/src/components/costs/energy-costs-change-last-thirty-days-card.tsx

```diff
@@ -1,13 +1,56 @@
+import { calculateCosts } from "@/components/dashboard/energy-cost";
+import { formatNumber } from "@energyleaf/lib";
 import { Card, CardContent, CardHeader, CardTitle } from "@energyleaf/ui/card";
 
-export default function EnergyCostsChangeLastThirtyDays() {
+function EnergyCostsChangeLastThirtyDays({ userData, energyData }) {
+    const now = new Date();
+    const lastThirtyDays = new Date(now.getFullYear(), now.getMonth(), now.getDate() - 30);
+    const sixtyDaysAgo = new Date(now.getFullYear(), now.getMonth(), now.getDate() - 60);
+
+    const lastMonthData = energyData.filter(
+        (data) => new Date(data.timestamp) >= sixtyDaysAgo && new Date(data.timestamp) < lastThirtyDays,
+    );
+    const thisMonthData = energyData.filter(
+        (data) => new Date(data.timestamp) >= lastThirtyDays && new Date(data.timestamp) < now,
+    );
+
+    if (lastMonthData.length < 30 || thisMonthData.length < 30) {
```

Comment:
Warum die < 30? (und in apps/web/src/components/costs/energy-costs-change-last-seven-days-card.tsx die < 7)? Ich verstehe, dass das die Anzahl der Tage in dem jeweiligen Intervall sind, aber warum genau ist das nötig?

Reply:
Es sollen halt 2 vollständige Monate Daten gesammelt sein, bis diese Dinger angezeigt werden
Problem ist halt, dass die Kacheln damit für die meisten Nutzer ja nie angezeigt werden

Februar ist da halt nicht mit drin :D

Wir sollten aus meiner Sicht solche Kacheln aber nicht während des ganzen Testzeitraums mit "Nicht genug Daten" anzeigen -> ich würde die dann lieber ausblenden.